src/InitialConditions/initial_conditions.jl

```diff
@@ -0,0 +1,763 @@
+
+    return (; T, p, q_tot, u, v)
+end
+
+"""
+    DryBaroclinicWave(; perturb = true)
+
+An `InitialCondition` with a dry baroclinic wave, and with an optional
+perturbation to the horizontal velocity.
+"""
+Base.@kwdef struct DryBaroclinicWave <: InitialCondition
+    perturb::Bool = true
+end
+
+function (initial_condition::DryBaroclinicWave)(params)
+    (; perturb) = initial_condition
+    function local_state(local_geometry)
+        thermo_params = CAP.thermodynamics_params(params)
+        (; z, lat, long) = local_geometry.coordinates
+        (; p, T_v, u, v) = baroclinic_wave_values(z, lat, long, params, perturb)
+        return LocalState(;
+            params,
+            geometry = local_geometry,
+            thermo_state = TD.PhaseDry_pT(thermo_params, p, T_v),
+            velocity = Geometry.UVVector(u, v),
+        )
+    end
+    return local_state
+end
+
+"""
+    MoistBaroclinicWave(; perturb = true)
+
+An `InitialCondition` with a moist baroclinic wave, and with an optional
+perturbation to the horizontal velocity.
+"""
+Base.@kwdef struct MoistBaroclinicWave <: InitialCondition
+    perturb::Bool = true
+end
+
+function (initial_condition::MoistBaroclinicWave)(params)
+    (; perturb) = initial_condition
+    function local_state(local_geometry)
+        thermo_params = CAP.thermodynamics_params(params)
+        (; z, lat, long) = local_geometry.coordinates
+        (; p, T, q_tot, u, v) =
+            moist_baroclinic_wave_values(z, lat, long, params, perturb)
+        return LocalState(;
+            params,
+            geometry = local_geometry,
+            thermo_state = TD.PhaseEquil_pTq(thermo_params, p, T, q_tot),
+            velocity = Geometry.UVVector(u, v),
+        )
+    end
+    return local_state
+end
+
+##
+## EDMF Test Cases
+##
+
+"""
+    hydrostatic_pressure_profile(; thermo_params, p_0, T, θ, q_tot)
+
+Solves the initial value problem `p'(z) = -g * ρ(z)`, given `p(0) = p_0`, either
+`T(z)` or `θ(z)`, and optionally also `q_tot(z)`. If `q_tot(z)` is not given, it
+is assumed to be 0.
+"""
+function hydrostatic_pressure_profile(;
+    thermo_params,
+    p_0,
+    T = nothing,
+    θ = nothing,
+    q_tot = nothing,
+)
+    FT = eltype(thermo_params)
+    grav = TD.Parameters.grav(thermo_params)
+
+    ts(p, z, ::Nothing, ::Nothing, _) = error("Either T or θ must be specified")
+    ts(p, z, T, θ, _) = error("Only one of T and θ can be specified")
+    ts(p, z, T, ::Nothing, ::Nothing) = TD.PhaseDry_pT(thermo_params, p, T(z))
+    ts(p, z, ::Nothing, θ, ::Nothing) = TD.PhaseDry_pθ(thermo_params, p, θ(z))
+    ts(p, z, T, ::Nothing, q_tot) =
+        TD.PhaseEquil_pTq(thermo_params, p, T(z), q_tot(z))
+    ts(p, z, ::Nothing, θ, q_tot) =
+        TD.PhaseEquil_pθq(thermo_params, p, θ(z), q_tot(z))
+    dp_dz(p, _, z) =
+        -grav * TD.air_density(thermo_params, ts(p, z, T, θ, q_tot))
+
+    z_span = (FT(0), FT(30e3)) # assume no one will ever need to go above 30km
```

Comment:
The grid information should probably be passed in, we have "high-top" simulations, so people may run into strange results with this assumption.

Reply:
Making the initial conditions depend on the grid information results in more complicated code, and it also throws off convergence studies. The profile to which we are interpolating should probably not depend on the model configuration.

Reply:
Also, the input profiles from `Cases.jl` don't work above 30 km anyway, which is why I used that value.

Reply:
@trontrytel's voted to keep this as is. Let's roll with it.